editor/plugins/gdextension/cpp_scons/template/addon/SConstruct_addon

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+
+# This file is for building as a Godot GDExtension.
+ARGUMENTS["precision"] = "double"
+env = SConscript("src/godot-cpp/SConstruct")
+
+# Add source files.
+env.Append(CPPPATH=["./,src/,__UPDIR_DOTS__/"])
+sources = Glob("*.cpp") + Glob("src/*.cpp") + Glob("__UPDIR_DOTS__/*.cpp")
+
+env.Append(CPPDEFINES=["GDEXTENSION"])
+
+bin_path = "bin/"
+extension_name = "__LIBRARY_NAME__"
+debug_or_release = "release" if env["target"] == "template_release" else "debug"
+
+if not "arch_suffix" in env:
+    env["arch_suffix"] = env["arch"]
+
+if env["target"] == "editor":
+    env.Append(CPPPATH=["__UPDIR_DOTS__/editor/"])
+    sources += Glob("__UPDIR_DOTS__/editor/*.cpp")
+
+if env["target"] in ["editor", "template_debug"]:
+    try:
+        doc_data = env.GodotCPPDocData("src/gen/doc_data.gen.cpp", source=Glob("doc_classes/*.xml"))
+        sources.append(doc_data)
+    except AttributeError:
+        print("Not including class reference as we're targeting a pre-4.3 baseline.")
```

Comment:
This test is not needed because this code will not be merged for pre-4.3 anyway. So it will never fail :)

Reply:
It could be the case that somebody uses Godot 4.4 and changes the GDExtension to target Godot 4.2 to support older Godot versions. But indeed, this could be removed if we don't care about that case. It was copied from the godot-cpp-template repo, added by this PR https://github.com/godotengine/godot-cpp-template/pull/39